src/main/webapp/app/guided-tour/guided-tour.service.ts

```diff
@@ -0,0 +1,367 @@
+import { ErrorHandler, Injectable } from '@angular/core';
+import { HttpClient, HttpResponse } from '@angular/common/http';
+import { cloneDeep } from 'lodash';
+import { JhiAlertService } from 'ng-jhipster';
+import { fromEvent, Observable, of, Subject } from 'rxjs';
+import { debounceTime } from 'rxjs/internal/operators';
+
+import { SERVER_API_URL } from 'app/app.constants';
+import { courseOverviewTour } from 'app/guided-tour/tours/course-overview-tour';
+import { GuidedTourSettings } from 'app/guided-tour/guided-tour-settings.model';
+import { ContentType, GuidedTour, Orientation, OrientationConfiguration, TourStep } from './guided-tour.constants';
+import { AccountService } from 'app/core';
+
+export type EntityResponseType = HttpResponse<GuidedTourSettings>;
+
+@Injectable()
+export class GuidedTourService {
+    public resourceUrl = SERVER_API_URL + 'api/guided-tour-settings';
+
+    public guidedTourCurrentStepStream: Observable<TourStep>;
+    public guidedTourOrbShowingStream: Observable<boolean>;
+    public currentTourSteps: TourStep[];
+    public guidedTourSettings: GuidedTourSettings;
+
+    private _guidedTourCurrentStepSubject = new Subject<TourStep>();
+    private _guidedTourOrbShowingSubject = new Subject<boolean>();
+    private _currentTourStepIndex = 0;
+    private _currentTour: GuidedTour | undefined;
+    private _onFirstStep = true;
+    private _onLastStep = true;
+    private _onResizeMessage = false;
+
+    private guidedTourNotification = new Subject<any>();
+
+    constructor(public errorHandler: ErrorHandler, private http: HttpClient, private jhiAlertService: JhiAlertService, private accountService: AccountService) {
+        this.getGuidedTourSettings();
```

Comment:
The constructor should only be used in Angular for initializing variables, everything else should be done in ngOnInit (also for injectable services).

Reply:
This comment is false, the lifecycle hooks are not called for services. I will therefore close this comment.